src/promo/entity/promo.entity.ts

```diff
@@ -0,0 +1,39 @@
+import type { IPromoCode } from 'promo/promo.interface';
+import { PromocodeState } from 'promo/promocode.enum';
+import { Column, Entity, PrimaryGeneratedColumn } from 'typeorm';
+
+@Entity({ name: 'promocode' })
+export class PromocodeEntity implements IPromoCode {
+  @PrimaryGeneratedColumn()
+    id: string;
+
+  @Column()
+    name: string;
+
+  @Column()
+    createdAt: string;
```

Comment:
typeorm provides autogenerated filed. Here we can use this pattern. We need to add appropriate decorator